flashlight/fl/dataset/SpanDataset.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "flashlight/fl/dataset/Dataset.h"
+
+#include <vector>
+
+namespace fl {
+
+/**
+ * A view into an underlying dataset with an offset and optional bounded length.
+ *
+ * The size of the `SpanDataset` is either specified for the size of the input dataset
+ * accounting for the offset.
+ *
+ * We have, for example `SpanDataset(ds, 13).get(i) == ds.get(13 + i)`
+ *
+ * Example:
+  \code{.cpp}
+  // Make a datasets
+  auto makeDataset = []() {
+    auto tensor = fl::rand({5, 4, 10});
+    std::vector<Tensor> fields{tensor};
```

Comment:
nit -- do you really need `makeDataset`?

Reply:
nope